docs/custom_container.md

```diff
@@ -0,0 +1,86 @@
+# Customizing the Azure DevOps job container
+
+The Model training and deployment pipeline uses a Docker container
+on the Azure Pipelines agents to provide a reproducible environment
+to run test and deployment code.
+ The image of the container
+`mcr.microsoft.com/mlops/python:latest` is built with this
+[Dockerfile](../environment_setup/Dockerfile).
+
+In your project you will want to build your own
+Docker image that only contains the dependencies and tools required for your
+use case. This image will be more likely smaller and therefore faster, and it
+will be totally maintained by your team. 
+
+## Provision an Azure Container Registry 
+
+An Azure Container Registry is deployed along your Azure ML Workspace to manage models.
+You can use that registry instance to store your MLOps container image as well, or
+provision a separate instance.
+
+## Create a Registry Service Connection
+
+Create a service connection to your Azure Container Registry:
+- As *Connection type*, select *Docker Registry*
+- As *Registry type*, select *Azure Container Registry*
+- As *Azure container registry*, select your Container registry instance
+- As *Service connection name*, enter `acrconnection`
+
+## Update the environment definition
+
+Modify the [Dockerfile](../environment_setup/Dockerfile) and/or the
+[ci_dependencies.yml](../diabetes_regression/ci_dependencies.yml) Conda
+environment definition to tailor your environment.
+
+If a package is available in a Conda package repository, then we recommend that
+you use the Conda installation rather than the pip installation. Conda packages
+typically come with prebuilt binaries that make installation more reliable.
+
+## Create a container build pipeline
+
+In your [Azure DevOps](https://dev.azure.com) project create a new build
+pipeline referring to the
+[./environment_setup/docker-image-pipeline.yml](../environment_setup/docker-image-pipeline.yml)
+pipeline definition in your forked repository.
+
+Create a pipeline variable named `agentImageName` and give it an appropriate
```

Comment:
This is confusing - where is this variable referenced?  Which pipeline variable - in the pipeline yml or task group variable? If I'm using ACR how do I know what name to use?

Reply:
Changed the approach to edit the default value in the YAML file rather than using a variable. This should be more straightforward.

Changed to:

Edit the [environment_setup/docker-image-pipeline.yml](../environment_setup/docker-image-pipeline.yml) file
and modify the string `'public/mlops/python'` with an name suitable to describe your environment,
e.g. `'mlops/diabetes_regression'`.